File: NHDK-Ten-Point-Round/Div. 3/Ten Point Round #37/C.py
n = eval(input())
x = []
y = []

ans = 0
for i in range(n):
    a,b = map(eval, input().split())
    x.append(a)
    y.append(b)

for i in range(n-1):
    ans += x[i]*y[i+1]
    ans -= y[i]*x[i+1]
ans += x[n-1]*y[0]
ans -= y[n-1]*x[0]

print(abs(ans)/2)

File: NHDK-Ten-Point-Round/Div. 3/Ten Point Round #37/C.cpp
#include <bits/stdc++.h>
#define ll long long
#define Andy8787_want_AC ios::sync_with_stdio(0),cin.tie(0);cout.tie(0);
using namespace std;

// 0分WTF
int main() {
    Andy8787_want_AC
    ll n;
    cin >> n;
    double x[n], y[n];
    for (ll i = 0; i < n; i++) {
        cin >> x[i] >> y[i];
    }

    double ans = 0;
    for (ll i = 0; i < n-1; i++) {
        ans += x[i]*y[i+1];
        ans -= y[i]*x[i+1];
    }
    ans += x[n-1]*y[0];
    ans -= y[n-1]*x[0];
    double a = (ans)/2;
    cout << abs(a);
}